src/flp/szk.rs

```diff
@@ -591,34 +610,56 @@ where
         ))
     }
 
-    /// Returns true if the verifier message indicates that the input from which
-    /// it was generated is valid.
+    pub(crate) fn merge_verifiers(
+        &self,
+        mut leader_share: SzkQueryShare<T::Field, SEED_SIZE>,
+        helper_share: SzkQueryShare<T::Field, SEED_SIZE>,
+    ) -> Result<SzkJointShare<SEED_SIZE>, SzkError> {
+        for (x, y) in leader_share
+            .flp_verifier
+            .iter_mut()
+            .zip(helper_share.flp_verifier)
+        {
+            *x += y;
+        }
+        if self.typ.decide(&leader_share.flp_verifier)? {
+            match (
+                leader_share.joint_rand_part_opt,
+                helper_share.joint_rand_part_opt,
+            ) {
+                (Some(leader_part), Some(helper_part)) => Ok(Some([leader_part, helper_part])),
+                (None, None) => Ok(None),
+                _ => Err(SzkError::InvalidState(
```

Comment:
As mentioned above, this error case is no longer about the prepare state, since this just depends on query shares.